### ISSUE TITLE
Bug with DateTimeFormat

### ISSUE THREAD

**Issue by mrbrianevans** [fullcalendar/temporal-polyfill#9]
Hi there, 
I've been using your polyfill and I think I've noticed an unexpected error that occurs when formatting a date time.
This error does not occur in the [spec playground](https://tc39.es/proposal-temporal/docs/index.html).

### To reproduce
Run
```javascript
Temporal.Instant.from(new Date().toISOString()).toLocaleString('en-GB', {dateStyle:'long'})
```
### Expected behaviour
In the [official spec playground](https://tc39.es/proposal-temporal/docs/index.html), the output is:
```json
"11 May 2022" 
```
### Actual behaviour
When run with this polyfill, it produces this error:
```
Uncaught TypeError: can't set option year when dateStyle is used
    buildFormat fromAndWith.ts:329
    toLocaleString intlFactory.ts:208
    <anonymous> pen.js:2
fromAndWith.ts:329
```

**Comment by hossameldeen** [comment]
I'm facing a similar issue:

This code:

```javascript
console.log(
  Temporal.PlainTime.from("15:30:37").toLocaleString([], {
    timeStyle: "short"
  })
);
```

results in this error:

```
TypeError: Can't set option hour when timeStyle is used
```

Codesandbox link: https://codesandbox.io/s/temporal-and-lodash-update-forked-dohu6n?file=/src/index.js

Perhaps [this issue](https://github.com/formatjs/formatjs/issues/2410) is relevant?

Also, it seems to work fine with Javascript `Intl.` API. And tried `PlainDateTime`, but it's the same issue.

![image](https://user-images.githubusercontent.com/6719916/225219888-8facad4c-4997-45ef-a523-83ebdc5a62b6.png)

**Comment by arshaw** [comment]
Confirmed, I'll fix this

**Comment by arshaw** [comment]
This has been fixed in [v0.2.0](https://github.com/fullcalendar/temporal-polyfill/releases/tag/v0.2.0)

Please lmk once you confirm. Thanks!
